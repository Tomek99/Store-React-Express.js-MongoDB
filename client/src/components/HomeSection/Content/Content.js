import React from "react";
import { Link } from "react-router-dom";
import styles from "./Content.module.scss";

function Content({ obj, slideIndex }) {
  return (() => {
    switch (slideIndex) {
      case 1:
        return (
          <div className={styles.content}>
            <header>{obj.title}</header>
            <p>{obj.text}</p>
            <a href="#menuSection">{obj.btnText}</a>
          </div>
        );
      case 2:
        return (
          <div className={styles.content}>
            <header>{obj.title}</header>
            <p>{obj.text}</p>
            <Link to="products/63dc02c4c2a0e09b2d62f8e8">{obj.btnText}</Link>
          </div>
        );

      default:
        return (
          <div className={`${styles.content} ${styles.content_3}`}>
            <header>{obj.title}</header>
            <p>{obj.text}</p>
            <Link to="products">{obj.btnText}</Link>
          </div>
        );
    }
  })();
}

export default Content;
