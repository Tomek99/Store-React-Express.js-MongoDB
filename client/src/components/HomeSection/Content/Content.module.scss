.content {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  position: absolute;
  top: 30%;
  left: 100px;
  margin: 20px;
  color: #fff;
  z-index: 1;
  opacity: 1;

  header {
    font-size: 5rem;
    text-transform: uppercase;
    font-weight: bold;
  }

  p {
    font-size: 2rem;
    max-width: 600px;
  }

  button {
    background-color: var(--main-color);
    text-transform: capitalize;
    color: #fff;
    border-radius: 8px;
    font-size: 1.7rem;
    padding: 12px 35px 12px 35px;
    margin-top: 10px;
    transition: 0.2s linear;
    cursor: pointer;

    &:hover {
      letter-spacing: 0.1rem;
    }
  }
}

.content_3 {
  display: flex;
  flex-direction: column;
  align-items: center;
  top: 100px;
  left: 0;
  right: 0;
}

@media (max-width: 1000px) {
  .content {
    align-items: center;
    justify-content: flex-start;
    top: 120px;
    left: 0;
    right: 0;

    header {
      text-align: center;
      font-size: 3rem;
    }

    p {
      margin-top: 10px;
      text-align: center;
      max-width: 500px;
      font-size: 1.5rem;
    }

    button {
      margin-top: 20px;
    }
  }
}

@media (max-width: 800px) {
  .content {
    align-items: center;
    justify-content: flex-start;
    top: 120px;
  }
}

@media (max-width: 350px) {
  .content {
    top: 25px;
  }
}
