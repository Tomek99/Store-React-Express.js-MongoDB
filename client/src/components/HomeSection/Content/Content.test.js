import { render } from "@testing-library/react";
import Content from "./Content";

test("should render Content component", () => {
  const obj = {
    title: "Test",
    text: "test test test test",
    btnText: "btnTest",
  };
  const { container } = render(<Content obj={obj} slideIndex={1} />);
  expect(container).toMatchSnapshot();
});
