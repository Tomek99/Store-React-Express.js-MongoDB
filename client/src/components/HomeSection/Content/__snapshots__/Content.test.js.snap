// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Content component 1`] = `
<div>
  <div
    class="content"
  >
    <header>
      Test
    </header>
    <p>
      test test test test
    </p>
    <a
      href="#menuSection"
    >
      btnTest
    </a>
  </div>
</div>
`;
